## Subhamay Dhara


## Intro to Machine Learning in Python

## 1

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
lst = [[35, 'Portugal', 94], [33, 'Argentina', 93], [30 , 'Brazil', 92]]
col = ["Age","Nationality","Overall"]

In [3]:
df = pd.DataFrame(lst, columns = col, index=[i for i in range(1,4)])
df

Age Nationality  Overall
1   35    Portugal       94
2   33   Argentina       93
3   30      Brazil       92

## 2

In [6]:
array  = np.array([[94, 89, 63],
                   [93, 92, 48],
                   [92, 94, 56]])
array[:2,1:]

array([[89, 63],
       [92, 48]])

## 3

Math is an inbuilt module in Python 

## 4

In [7]:
df.shape

(3, 3)

## 5

In [8]:
S = [['him', 'sell'], [90, 28, 43]]
S[0][1][1]

'e'

## 6

In [9]:
my_tuppy = (1,2,5,8)
my_tuppy[2] = 6

TypeError: 'tuple' object does not support item assignment

## 7

In [12]:
y = [(2, 4), (7, 8), (1, 5, 9)]
x = y[1][1]
x1 = y[1][-1]

print(x)
print(x1)

8
8


## 8

Dictionary data structure is unorderly in Python

## 9

Pandas data structure with dimention (100, 3) has 3 features and 100 observations.

## 10

In [15]:
df.iloc[0:3]

Age Nationality  Overall
1   35    Portugal       94
2   33   Argentina       93
3   30      Brazil       92

## 11

In [16]:
data = pd.read_csv("FoodBalanceSheets_E_Africa_NOFLAG.csv", encoding="latin-1")
data

Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     1000 persons  38924.00   
1                    Domestic supply quantity      1000 tonnes      0.00   
2               Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3      Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4          Fat supply quantity (g/capita/day)     g/capita/day     80.06   
...                                       ...              ...       ...   
60938                                    Food      1000 tonnes     42.00   
60939     Food supply quantity (kg/capita/yr)               kg      3.06   
60940           Food supply (kcal/capita/day)  kcal/capita/day      3.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day      0.10   
60942      Fat supply quantity (g/capita/day)     g/capita/day      0.04   

          Y2015     Y2016     Y2017     Y2018  
0      39728.00  40551.00  41389.00  42228.00  
1          0.00      0.00      0.00      0.00  
2       3379.00   3372.00   3341.00   3322.00  
3         94.35     94.72     92.82     91.83  
4         79.36     77.40     80.19     77.28  
...         ...       ...       ...       ...  
60938     46.00     33.00     19.00     16.00  
60939      3.33      2.35      1.33      1.08  
60940      4.00      3.00      1.00      1.00  
60941      0.11      0.08      0.04      0.04  
60942      0.05      0.03      0.02      0.01  

[60943 rows x 12 columns]

In [17]:
data.groupby(["Item"]).sum()[["Y2014", "Y2017"]]

Y2014      Y2017
Item                                     
Alcohol, Non-Food      2403.00    2348.00
Alcoholic Beverages  102410.11   95581.06
Animal Products       11935.65   11547.65
Animal fats          209460.54  269617.53
Apples and products    9499.23   10198.90
...                        ...        ...
Vegetables, Other    155038.96  157752.59
Vegetal Products     107145.19  107655.20
Wheat and products   232670.13  240047.62
Wine                   4497.36    4178.02
Yams                 200396.96  229174.59

[119 rows x 2 columns]

Animal fats produce in 2014 is 209460.54 and in 2017 is 269617.53

## 12

In [20]:
mean = np.mean(data.Y2015)
mean_3 = np.round(mean, 3)
mean_3

135.236

In [22]:
std = np.std(data.Y2015)
std_3 = np.round(std, 3)
std_3

1603.39

mean = 135.236 & std = 1603.39

## 13

In [23]:
data.isna().sum()["Y2016"]

1535

## 14

In [25]:
data.corr()["Element Code"]

Area Code      -0.000209
Item Code      -0.024683
Element Code    1.000000
Y2014           0.024457
Y2015           0.023889
Y2016           0.023444
Y2017           0.024254
Y2018           0.024279
Name: Element Code, dtype: float64

Highest Correlation with Element Code is 0.024457 in 2014

## 15

In [26]:
data.groupby(["Element"]).sum().loc["Import Quantity"]

Area Code         688174.00
Item Code       13795966.00
Element Code    28834929.00
Y2014             274144.48
Y2015             267018.46
Y2016             286582.78
Y2017             294559.09
Y2018             287997.09
Name: Import Quantity, dtype: float64

Highest Import Quantity is 294559.09 in 2017

## 16

In [27]:
data.groupby(["Element"]).sum().loc["Production","Y2014"]

1931287.7500000005

Total no of sum of Production in 2014 is 1931287.7500000005

## 17

In [28]:
elements = data[["Y2018", "Element"]].groupby(["Element"]).sum()
elements.sort_values("Y2018", ascending=False)

Y2018
Element                                           
Domestic supply quantity                2161192.10
Production                              2075072.89
Food                                    1303841.28
Total Population - Both sexes           1140605.00
Food supply (kcal/capita/day)            455261.00
Processing                               308429.00
Import Quantity                          287997.09
Feed                                     233489.68
Export Quantity                          181594.80
Losses                                   163902.00
Other uses (non-food)                     91300.97
Food supply quantity (kg/capita/yr)       49056.85
Residuals                                 34864.00
Seed                                      25263.14
Stock Variation                           20577.91
Protein supply quantity (g/capita/day)    11833.56
Fat supply quantity (g/capita/day)        10258.69
Tourist consumption                          90.00

Domestic supply quantity had the highest sum in 2018

## 18

In [30]:
elements = data[["Y2018", "Element"]].groupby(["Element"]).sum()
elements.sort_values("Y2018")

Y2018
Element                                           
Tourist consumption                          90.00
Fat supply quantity (g/capita/day)        10258.69
Protein supply quantity (g/capita/day)    11833.56
Stock Variation                           20577.91
Seed                                      25263.14
Residuals                                 34864.00
Food supply quantity (kg/capita/yr)       49056.85
Other uses (non-food)                     91300.97
Losses                                   163902.00
Export Quantity                          181594.80
Feed                                     233489.68
Import Quantity                          287997.09
Processing                               308429.00
Food supply (kcal/capita/day)            455261.00
Total Population - Both sexes           1140605.00
Food                                    1303841.28
Production                              2075072.89
Domestic supply quantity                2161192.10

Protein supply quantity (g/capita/day) had the 3rd lowest sum in 2018

## 19

In [31]:
algeria = data[data["Area"]=="Algeria"]
algeria.groupby("Element").sum().loc["Import Quantity","Y2018"]

36238.29

Total Import Quantity in Algeria in 2018 is 36238.29

## 20

In [35]:
data["Area"].nunique()

49

Total no of Unique Countries is 49